pyqtgraph/graphicsItems/PColorMeshItem.py

```diff
@@ -0,0 +1,251 @@
+        if len(args)>0:
+            self.setData(*args)
+
+
+    def _prepareData(self, args):
+        """
+        Check the shape of the data.
+        Return a set of 2d array x, y, z ready to be used to draw the picture.
+        """
+
+        # User didn't specified data
+        if len(args)==0:
+
+            self.x = None
+            self.y = None
+            self.z = None
+            
+        # User only specified z
+        elif len(args)==1:
+            # If x and y is None, the polygons will be displaced on a grid
+            x = np.arange(0, args[0].shape[0]+1, 1)
+            y = np.arange(0, args[0].shape[1]+1, 1)
+            self.x, self.y = np.meshgrid(x, y, indexing='ij')
+            self.z = args[0]
+
+        # User specified x, y, z
+        elif len(args)==3:
+
+            # Shape checking
+            if args[0].shape[0] != args[2].shape[0]+1 or args[0].shape[1] != args[2].shape[1]+1:
+                raise ValueError('The dimension of x should be one greater than the one of z')
+            
+            if args[1].shape[0] != args[2].shape[0]+1 or args[1].shape[1] != args[2].shape[1]+1:
+                raise ValueError('The dimension of y should be one greater than the one of z')
+        
+            self.x = args[0]
+            self.y = args[1]
+            self.z = args[2]
+
+        else:
+            ValueError('Data must been sent as (z) or (x, y, z)')
+
+
+    def setData(self, *args):
+        """
+        Set the data to be drawn.
+
+        Parameters
+        ----------
+        x, y : np.ndarray, optional, default None
+            2D array containing the coordinates of the polygons
+        z : np.ndarray
+            2D array containing the value which will be maped into the polygons
+            colors.
+            If x and y is None, the polygons will be displaced on a grid
+            otherwise x and y will be used as polygons vertices coordinates as:
+
+            (x[i+1, j], y[i+1, j])           (x[i+1, j+1], y[i+1, j+1])
+                                +---------+
+                                | z[i, j] |
+                                +---------+
+                (x[i, j], y[i, j])           (x[i, j+1], y[i, j+1])
+            "ASCII from: https://matplotlib.org/3.2.1/api/_as_gen/
+                         matplotlib.pyplot.pcolormesh.html".
+
+        """
+
+        # Prepare data
+        cd = self._prepareData(args)
+
+        # Has the view bounds changed
+        shapeChanged = False
+        if self.qpicture is None:
+            shapeChanged = True
+        elif len(args)==1:
+            if args[0].shape[0] != self.x[:,1][-1] or args[0].shape[1] != self.y[0][-1]:
+                shapeChanged = True
+        elif len(args)==3:
+            if np.any(self.x != args[0]) or np.any(self.y != args[1]):
+                shapeChanged = True
+
+        profile = debug.Profiler()
+
+        self.qpicture = QtGui.QPicture()
+        p = QtGui.QPainter(self.qpicture)
+        
```

Comment:
totally optional here, but I've been experimenting on my own, but consider setting the anti-aliassing render hint

```python
        p.setRenderHint(QtGui.QPainter.Antialiasing)
```

With this set, the lines were actually visible on my screen; admittedly that may not be what you want.